src/frontend/js/bulk-actions.ts

```diff
@@ -103,4 +106,29 @@ window.addEventListener("load", () => {
     // Submit form and execute bulk action
     form.submit();
   }
+
+  function hasTranslation(): boolean {
+    // checks if at least one of the selected pages has a translation for the current language
+    let urlPath = window.location.pathname;
+    // remove trailing slash from url path
+    urlPath = urlPath.replace(/\/$/, '');
+    // extract current language slug from url path
+    let languageSlug = urlPath.substr(urlPath.lastIndexOf('/') + 1);
+    // retrieve all checked input elements
+    let selectedItems = selectItems
+      .filter(isInputElement)
+      .filter(inputElement => inputElement.checked === true);
+    for (let translation of selectedItems){
+      // select for each checked input the current language state
+      let translationState = translation
+        .closest("tr")
+        .querySelector('.lang-grid')
+        .querySelector(`.${languageSlug}`);
+      if (translationState.querySelector(".no-trans") === null)
+        // if at least one page does not contain a no-trans-class for this language
+        // then there exists a translation for the given page-language-combination
+        return true;
+    }
+    return false;
```

Comment:
```suggestion
    // check if at least one input element has a translation
    return selectItems
      .filter(isInputElement)
      .filter(inputElement => inputElement.checked)
      .some(inputElement => {
        return inputElement
          .closest("tr")
          .querySelector(`.lang-grid .${languageSlug} .no-trans`) === null;
      });
```
This can be simplified a little bit...

Reply:
I like this `some()` function, I didn't know it yet :rocket: